### PR TITLE
[WIP] ref: part 6 of #3733

### PR DESCRIPTION
**Pull request by williamFalcon**
ref: part 6 of #3733

### CODE REVIEW

File: pytorch_lightning/accelerators/ddp_backend.py [hunk 1]
```diff
@@ -149,15 +261,7 @@ def set_world_ranks(self, process_idx):
...
-        gpu_idx = process_idx
-
-        # when using ddp, the master process (proc 0) continues running as the main one
-        # this means that the local rank will always be 0
-        # (even if cuda visible devices has other visible gpus)
-        # this means that the master process needs to pull the 0th visible index as the device number
-        if is_master:
-            available_gpus = os.environ['CUDA_VISIBLE_DEVICES'].split(',')
-            gpu_idx = int(available_gpus[self.trainer.local_rank])
+        gpu_idx = int(os.environ.get('PL_DDP_PID', process_idx))
```
Comment (ananthsub) [inline pytorch_lightning/accelerators/ddp_backend.py:264]: `is_master` is never used here. it seems like ddp_train doesn't need the self.mode == 'ddp' check anymore

File: pytorch_lightning/accelerators/ddp_backend.py [hunk 2]
```diff
@@ -127,14 +137,116 @@ def __ddp_script_mode_setup(self):
...
+        sleep(2)
+
     def train(self):
         model = self.trainer.model
         if self.mode == 'ddp':
-            results = self.ddp_train_tmp(process_idx=self.task_idx, mp_queue=None, model=model, is_master=True)
-            del os.environ['WORLD_SIZE']
+            results = self.ddp_train(process_idx=self.task_idx, mp_queue=None, model=model, is_master=True)
+            if 'WORLD_SIZE' in os.environ:
+                del os.environ['WORLD_SIZE']
```
Comment (ananthsub) [inline pytorch_lightning/accelerators/ddp_backend.py:148]: n00b question: should this happen in a teardown step?

File: pytorch_lightning/accelerators/ddp_backend.py (hunk 1 above)
Comment (williamFalcon) [reply pytorch_lightning/accelerators/ddp_backend.py:264]: Good catch! 
might be a weird merge artifact... let me check.

File: pytorch_lightning/accelerators/ddp_backend.py (hunk 2 above)
Comment (williamFalcon) [reply pytorch_lightning/accelerators/ddp_backend.py:148]: probably haha.

i'm not sure what will happen if we move it though. I want to lock down the tests then we can see where it goes

**Pull request closed without merging**